staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4876,6 +4880,10 @@ type PodStatus struct {
 	// e.g. 'Evicted'
 	// +optional
 	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// The generation of the pod observed at the point this pod status is being reported.
```

Comment:
because this is a new field which could be unset (when talking to older API servers), it's worth documenting this as a conditional, e.g.:

```
If set, this represents the .metadata.generation that the pod status was set based upon.
```

and if we're going to clear this field until PodObservedGenerationTracking is enabled, this should also document this is an alpha-level field that requires the PodObservedGenerationTracking feature to be enabled, e.g.:

```
This is an alpha field. Enable PodObservedGenerationTracking to be able to use this field.
```

Reply:
It's not a huge deal, but the *most* common location in status for ObservedGeneration is as the first field in the struct... I would recommend moving this to the top of the PodStatus struct (here and in core/types.go)

This does actually impact serialization order, so when serializing, that would output this field first, which *sort* of makes sense if the meaning of this field is "this is the metadata.generation value the rest of the status stuff is *for*"

Reply:
Done